backend/src/core/services/job.service.ts

```diff
@@ -14,14 +14,40 @@ import { ListService } from '.'
 const createJob = async ({
   campaignId,
   rate,
+  scheduledTiming = undefined,
 }: {
   campaignId: number
   rate: number
+  scheduledTiming?: Date
 }): Promise<number | undefined> => {
+  if (!scheduledTiming) {
+    // if its not a scheduled campaign, just throw in current timing. will not affect main flow of logic
+    scheduledTiming = new Date()
+  }
+
+  //format scheduled Timing to a formatted string
+  // i think i needa offset 8 hrs man
+  const stringTiming =
```

Comment:
cmiiw but u can just pass the `Date` object in directly without having to convert it to a string

Reply:
tried this before settling on this implementation, if pass as date object, need to manually offset 8 hours ( think its due to DB being localized as GMT+8, so db auto adds 8 hours to timestamps passed in). but if i pass in as a UTC string as cast it as timestamp, then DB's auto convert will add the GMT timing to it

Reply:
Eh no leh the Date object should contain timezone information, can u show us this issue in our pair review?

Reply:
wait, im trying it rn and it seems to be working now. i was pressure sure that the offset was screwing me over back when i tried it

Reply:
fixed